src/SFA.DAS.AssessorService.Application/Handlers/Learner/ImportLearnerDetailHandler.cs

```diff
@@ -0,0 +1,200 @@
+﻿using MediatR;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.AssessorService.Api.Types.Models;
+using SFA.DAS.AssessorService.Application.Interfaces;
+using SFA.DAS.AssessorService.Domain.Entities;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.AssessorService.Application.Handlers.Learner
+{
+    public class ImportLearnerDetailHandler : IRequestHandler<ImportLearnerDetailRequest, ImportLearnerDetailResponse>
+    {
+        private readonly IIlrRepository _ilrRepository;
+        private readonly ICertificateRepository _certificateRepository;
+        private readonly ILogger<ImportLearnerDetailHandler> _logger;
+
+        public ImportLearnerDetailHandler(IIlrRepository ilrRepository, ICertificateRepository certificateRepository, ILogger<ImportLearnerDetailHandler> logger)
+        {
+            _ilrRepository = ilrRepository;
+            _certificateRepository = certificateRepository;
+            _logger = logger;
+        }
+
+        public async Task<ImportLearnerDetailResponse> Handle(ImportLearnerDetailRequest request, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("Handling Import Learner Detail Request");
+
+            return CheckMissingMandatoryFields(request) ?? new ImportLearnerDetailResponse
+            {
+                Result = await HandleRequest(request, cancellationToken)
+            };
+        }
+
+        private async Task<string> HandleRequest(ImportLearnerDetailRequest request, CancellationToken cancellationToken)
+        {
```

Comment:
would be nice to simplify this code, but might be one for a tech debt task

Reply:
Updated to reduce nesting and removed duplicated return values.